idaes/generic_models/properties/core/coolprop/coolprop_wrapper.py

```diff
@@ -0,0 +1,486 @@
+
+    This class is intended to be used in theromphysical property definition
+    dicts and directs the properties framework to use forms and parameter data
+    from the CoolProp libraries (if available). This requires that the user
+    have CoolProp installed locally.
+    """
+    _cached_components = {}
+
+    @staticmethod
+    def get_parameter_value(comp_name, param):
+        """
+        Method to retrieve tuples of (value, units) for the specified property
+        and component for the critical state section of the CoolProp json
+        format.
+
+        Args:
+            comp_name - name of componet for which to retirve parameters
+            param - name of parameter to return value and units
+
+        Returns:
+            tuple of parameter (value, units)
+        """
+        map_tuple = name_map.get(param, None)
+        if map_tuple is None:
+            raise BurntToast(
+                "Unrecognized property name in CoolProp wrapper. Please "
+                "contact the IDAES developers with this bug.")
+
+        if map_tuple[1] == "CRITICAL":
+            ptuple = CoolPropWrapper._get_critical_property(
+                comp_name, map_tuple[0])
+        elif map_tuple[1] == "EOS":
+            ptuple = CoolPropWrapper._get_eos_property(
+                comp_name, map_tuple[0])
+        else:
+            raise BurntToast(
+                "Unrecognized argument in CoolProp wrapper. Please contact "
+                "the IDAES developers with this bug.")
+
+        return ptuple
+
+    @staticmethod
+    def flush_cached_components():
+        """
+        Method to clear cached component parameter data. This sets
+        _cached_components to an empty dict.
+
+        Args:
+            None
+
+        Returns:
+            None
+        """
+        CoolPropWrapper._cached_components = {}
+
+    # -------------------------------------------------------------------------
+    # Pure component property sub-classes
+    class dens_mol_liq_comp():
+        """
+        Class for calculating liquid molar density using CoolProp forms and
+        parameters.
+        """
+
+        @staticmethod
+        def build_parameters(cobj):
+            cname = cobj.local_name
+            cdict = CoolPropWrapper._get_component_data(cname)
+
+            # 8-Dec-21: CoolProp only uses rhoLnoexp for liquid density
+            nlist, tlist = CoolPropWrapper._get_param_dicts(
+                    cname,
+                    cdict,
+                    "dens_mol_liq_comp",
+                    "rhoL",
+                    ["rhoLnoexp"])
+
+            cforms.parameters_nt_sum(cobj, "dens_mol_liq_comp", nlist, tlist)
+
+        @staticmethod
+        def return_expression(b, cobj, T):
+            return cforms.expression_nonexponential(
+                cobj, "dens_mol_liq_comp", T, cobj.dens_mol_crit)
+
+    class enth_mol_liq_comp():
+        """
+        Class for calculating liquid molar enthalpy using CoolProp forms and
```

Comment:
Style nitpick: Instead of saying "class for doing X" just say "do X". We can see it's a class. e.g.
```
Calculate liquid molar enthalpy using CoolProp forms and ...
```